frontend/src/app/modules/main/challenge/challenge.component.ts

```diff
@@ -10,9 +10,15 @@ import { getLanguageIconUrl } from '@shared/utils/language-icons';
 export class ChallengeComponent {
     @Input() challenge: ChallengePreview;
 
+    @Output() starChallenge = new EventEmitter<ChallengePreview>();
+
     public challengeStars = 0;
 
     public challengePositiveFeedbacksPercent = 0;
 
     public getLanguageIconUrl = getLanguageIconUrl;
+
+    public starChange() {
+        this.starChallenge.emit(this.challenge);
+    }
```

Comment:
you can move out logic with starService here from filtering section component and avoid using Output